### PR TITLE
Added confirmation for registration

### PR DESCRIPTION
**Pull request by evrusha**

### CODE REVIEW

**Review by Terapool — APPROVED** [review]

File: app/views/user_mailer/registration_confirmation.html.slim [hunk 1]
```diff
@@ -0,0 +1,5 @@
+p Hi #{@user.name},
+
+p Thanks for registering! To confirm your registration click the URL below.
+
+href="http://localhost:3000/sign_up/confirm_email?token=#{@user.confirm_token}"
```
Comment (ritamma) [inline app/views/user_mailer/registration_confirmation.html.slim:5]: Will we change the localhost to the real URL later?

Comment (evrusha) [reply app/views/user_mailer/registration_confirmation.html.slim:5]: I think we'll change it later. Because there is a moment with https on heroku.

**Review by ritamma — APPROVED** [review]

File: app/models/user.rb [hunk 2]
```diff
@@ -51,9 +51,21 @@ def reset_password!(password)
     save!
   end
 
+  def email_activate
```
Comment (fooooxmr) [inline app/models/user.rb:54]: It's better to use ! in the end of the method. This method changes the object.

File: app/views/user_mailer/registration_confirmation.text.slim [hunk 3]
```diff
@@ -0,0 +1,5 @@
+p Hi #{@user.name},
+
+p Thanks for registering! To confirm your registration click the URL below.
+
+href="http://localhost:3000/sign_up/confirm_email?token=#{@user.confirm_token}"
```
Comment (fooooxmr) [inline app/views/user_mailer/registration_confirmation.text.slim:5]: In the link there should be ENV variable with URL for the frontend.Locally this ENV will be localhost, but on production and test server different.

File: app/controllers/sign_up_controller.rb [hunk 4]
```diff
@@ -11,16 +12,30 @@ def create
   def login
     @user = User.find_by(email: params[:email])
     if @user&.authenticate(params[:password])
-      token = JsonWebToken.encode(name: @user.name, email: @user.email,
-                                  description: @user.description, phone: @user.phone,
-                                  gender: @user.gender, birthday: @user.birthday.to_s,
-                                  exp: 1.hour.from_now.to_i)
-      render json: { jwt: token }, status: :ok
+      if @user.email_confirmed
```
Comment (fooooxmr) [inline app/controllers/sign_up_controller.rb:15]: I guess, you can use `@user.email_confirmed?` and it will also work.

**Review by fooooxmr — APPROVED** [review]

File: app/models/user.rb (hunk 2 above)
Comment (evrusha) [reply app/models/user.rb]: Fixed.

File: app/views/user_mailer/registration_confirmation.text.slim (hunk 3 above)
Comment (evrusha) [reply app/views/user_mailer/registration_confirmation.text.slim]: Fixed.

File: app/controllers/sign_up_controller.rb (hunk 4 above)
Comment (evrusha) [reply app/controllers/sign_up_controller.rb]: Fixed.

**Pull request merged**